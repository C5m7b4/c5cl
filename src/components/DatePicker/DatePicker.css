.c5-date-picker {
  position: relative;
}

.c5-date-picker * {
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5-dp-input {
  width: 150px;
  height: 35px;
  overflow: hidden;
  border-radius: 20px;
}

.c5-dp-input:focus {
  outline: none;
}

.c5-dp-input input {
  width: 125%;
  background: #f5f5f5;
  border: 0;
  outline: none;
  height: 35px;
  text-align: center;
  cursor: pointer;
  user-select: none;
}

.mdp-container {
  position: absolute;
  left: 0;
  top: 40px;
  width: 300px;
  min-height: 150px;
  background: #fff;
  box-shadow: 10px 10px 40px rgba(0, 0, 0, 0.3);
  border-radius: 20px;
  overflow: hidden;
  padding: 25px 30px;
}

.mdp-head {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
}

.mdp-head-date {
  text-align: center;
}

.mdp-button {
  float: left;
  width: 45px;
  height: 100%;
  box-sizing: border-box;
  position: relative;
}

.mdp-head-button-inner:hover span {
  border-color: #555;
}

.mdp-head-button-inner:hover {
  cursor: pointer;
  background: #eee;
}

.mdp-head-button-inner {
  float: left;
  height: 35px;
  width: 35px;
  background: #f4f4f4;
  border-radius: 100%;
  line-height: 35px;
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  margin-left: -17px;
  margin-top: -17px;
}

.mdp-head-button-left-arrows:after,
.mdp-head-button-right-arrows:after,
.mdp-head-button-right-arrow,
.mdp-head-button-left-arrow,
.mdp-head-button-left-arrows,
.mdp-head-button-right-arrows {
  display: block;
  float: left;
  width: 6px;
  height: 6px;
  border-left: 2px solid #888;
  border-bottom: 2px solid #888;
  position: absolute;
}

.mdp-head-button-right-arrow,
.mdp-head-button-right-arrows,
.mdp-head-button-left-arrow,
.mdp-head-button-left-arrows {
  transform: rotate(45deg);
  left: 50%;
  top: 50%;
  margin-left: -2px;
  margin-top: -4px;
}

.mdp-head-button-right-arrows,
.mdp-head-button-right-arrow {
  transform: rotate(225deg);
  margin-left: -4px;
}

.mdp-head-button-right-arrows:after,
.mdp-head-button-left-arrows:after {
  content: '';
}

.mdp-head-button-left-arrows {
  margin-left: -5px;
}

.mdp-head-button-right-arrows {
  margin-left: -2px;
}

.mdp-head-button-right-arrows:after,
.mdp-head-button-left-arrows:after {
  left: 3px;
  top: -5px;
}

.cc-month,
.cc-year,
.cch-name,
.cc-body,
.ckc-day span,
.cdc-day,
.c-day-container,
.c-container {
  position: relative;
  display: block;
  float: left;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.cch-name {
  width: 14.285%;
  height: 30px;
  line-height: 30px;
  font-weight: 700;
  color: #666;
  font-size: 14px;
  text-align: center;
}

.c-container {
  width: 100%;
  height: 100%;
}

.cc-head {
  height: 30px;
  width: 100%;
  margin-top: 10px;
}

.cch-name {
  width: 14.285%;
  height: 30px;
  line-height: 30px;
  font-weight: 700;
  color: #666;
  font-size: 9px;
  text-align: center;
}

.cc-body {
  height: 200px;
  width: 100%;
}

.c-day-container {
  width: 14.285%;
  height: 16.6666%;
}

.cdc-day {
  width: 100%;
  height: 100%;
  font-size: 12px;
  font-weight: 300;
  color: #444;
  text-align: center;
}

.cdc-day span {
  width: 100%;
  height: 100%;
  font-size: 12px;
  font-weight: 300;
  color: #444;
}

.cdc-day span:hover {
  cursor: pointer;
  background: #99db6f;
  height: 35px;
  width: 35px;
  padding: 10px;
}

.cdc-day span {
  width: 30px;
  height: 30px;
  margin-top: -15px;
  margin-left: -15px;
  left: 50%;
  top: 50%;
  font-weight: 400;
  border-radius: 100%;
  line-height: 30px;
}

.c-day-container.disabled {
  pointer-events: none;
}

.c-day-container.disabled .cdc-day span {
  color: #ddd;
}

.c-day-container.disabled .cdc-day span {
  background: #fff !important;
}

.c-day-container.highlight .cdc-day span {
  background: #e68b3c;
  padding: 10px;
}

.c-day-container.highlight-green .cdc-day span {
  /* background: #d4e2cb; */
  background: #99db6f;
  padding: 10px;
}
