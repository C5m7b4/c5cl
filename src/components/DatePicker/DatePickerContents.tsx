import React, { useState } from 'react';
import Tooltip from '../Tooltip';
import { convertToDateFormat, months, getMonthDetails } from './core';

export interface DatePickerContentsProps {
  componentId: string;
  top: number;
  left: number;
  monthDetails: any;
  setDefaultDate: (d: Date) => void;
  defaultDate: Date;
  onDateClick: (d: any) => void;
}

type CalendarDay = {
  date: number;
  day: number;
  dayString: string;
  month: number;
  timestamp: number;
};

const DatePickerContents = (props: DatePickerContentsProps) => {
  const {
    componentId,
    top,
    left,
    monthDetails,
    setDefaultDate,
    defaultDate,
    onDateClick,
  } = props;

  const isCurrentDay = (day: CalendarDay) => {
    if (day.month === 0 && day.date === defaultDate.getDate()) {
      return true;
    } else {
      return false;
    }
  };

  const isSelectedDay = (day: CalendarDay) => {
    return day.timestamp === defaultDate.getTime();
  };

  const handleYearChange = (increment: number) => {
    const month = defaultDate.getMonth() + 1;
    const year = defaultDate.getFullYear() + increment;
    const day = defaultDate.getDate();
    const newDate = convertToDateFormat(
      `${year.toString()}-${
        month.toString().length === 1
          ? '0' + month.toString()
          : month.toString()
      }-${day.toString().length === 1 ? '0' + day.toString() : day.toString()}`
    );
    setDefaultDate(new Date(newDate));
  };

  const handleMonthChange = (increment: number) => {
    let month = defaultDate.getMonth() + 1 + increment;
    let year = defaultDate.getFullYear();
    const day = defaultDate.getDate();
    if (month === 0) {
      month = 12;
      year--;
    } else if (month > 12) {
      year++;
      month = 1;
    }

    const newDate = convertToDateFormat(
      `${year.toString()}-${
        month.toString().length === 1
          ? '0' + month.toString()
          : month.toString()
      }-${day.toString().length === 1 ? '0' + day.toString() : day.toString()}`
    );
    setDefaultDate(new Date(newDate));
  };

  const renderCalendar = () => {
    if (typeof monthDetails === 'undefined') return;
    if (monthDetails.length === 0) {
      return <div>There are no results</div>;
    }
    let days = monthDetails.map((day: any, index: number) => {
      return (
        <div
          className={`c-day-container ${day.month !== 0 ? 'disabled' : ''} ${
            isCurrentDay(day) ? ' highlight' : ''
          } ${isSelectedDay(day) ? ' hightlight-green' : ''}`}
          key={index}
        >
          <div className="cdc-day">
            <span onClick={() => onDateClick(day)}>{day.date}</span>
          </div>
        </div>
      );
    });

    return (
      <div className="c-container">
        <div className="cc-head">
          {['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'].map((d, i) => (
            <div key={i} className="cch-name">
              {d}
            </div>
          ))}
        </div>
        <div className="cc-body">{days}</div>
      </div>
    );
  };

  return (
    <div
      className="mdp-container"
      style={{ position: 'absolute', top: `${top}px`, left: `${left}px` }}
    >
      <div className="mdp-content-container">
        <div className="mdp-head">
          <div className="mdp-button">
            <div
              id={`${componentId}-ly`}
              className="mdp-head-button-inner"
              onClick={() => handleYearChange(-1)}
            >
              <Tooltip position="bottom" message="Subtract Year" theme="dark">
                <span className="mdp-head-button-left-arrows"></span>
              </Tooltip>
            </div>
          </div>
          <div className="mdp-button">
            <div
              className="mdp-head-button-inner"
              id={`${componentId}-lm`}
              onClick={() => handleMonthChange(-1)}
            >
              <Tooltip position="bottom" message="Subtract month" theme="dark">
                <span className="mdp-head-button-left-arrow"></span>
              </Tooltip>
            </div>
          </div>
          <div className="mdp-head-date">
            <div className="mdp-head-year">{defaultDate.getFullYear()}</div>
            <div className="mdp-head-month">
              {months[defaultDate.getMonth()]}
            </div>
          </div>
          <div className="mdp-button">
            <div
              className="mdp-head-button-inner"
              id={`${componentId}-nm`}
              onClick={() => handleMonthChange(1)}
            >
              <Tooltip position="bottom" message="Add month" theme="dark">
                <span className="mdp-head-button-right-arrow"></span>
              </Tooltip>
            </div>
          </div>
          <div className="mdp-button">
            <div
              className="mdp-head-button-inner"
              id={`${componentId}-ny`}
              onClick={() => handleYearChange(1)}
            >
              <Tooltip position="bottom" message="Add year" theme="dark">
                <span className="mdp-head-button-right-arrows"></span>
              </Tooltip>
            </div>
          </div>
        </div>
      </div>

      <div className="mdp-body">{renderCalendar()}</div>
    </div>
  );
};

export default DatePickerContents;
