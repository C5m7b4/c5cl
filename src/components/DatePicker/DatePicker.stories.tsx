import React from 'react';
import { Story, Meta } from '@storybook/react';

import DatePicker, { DatePickerProps } from './DatePicker';

export default {
  title: 'Pickers/DatePicker1',
  component: DatePicker,
} as Meta;

const Template: Story<DatePickerProps> = (args) => <DatePicker {...args} />;

export const DatePickerExample = Template.bind({});
DatePickerExample.args = {
  date: new Date(),
};
