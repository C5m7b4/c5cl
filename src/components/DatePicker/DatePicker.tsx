import React, { useState, useEffect, useRef } from 'react';
import ReactDOM from 'react-dom';
import { convertToDateFormat, months, getMonthDetails } from './core';
import Tooltip from '../Tooltip';

import './DatePicker.css';

export interface DatePickerProps {
  date: Date;
  onChange: (d: Date) => void;
}

const DatePicker = (props: DatePickerProps) => {
  const [defaultDate, setDefaultDate] = useState(props.date);
  const [showDatePicker, setShowDatePicker] = useState(false);
  const [monthDetails, setMonthDetails] = useState<any>();

  const dateRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    setInput();
    setMonthDetails(
      getMonthDetails(defaultDate.getFullYear(), defaultDate.getMonth())
    );
  }, [defaultDate]);

  const setInput = () => {
    if (dateRef.current) {
      const dateString = convertToDateFormat(new Date(defaultDate).toString());
      console.log('dateString', dateString);
      dateRef.current.value = dateString;
    }
  };

  const handleDateClick = () => {
    setShowDatePicker(true);
    // @ts-ignore
    window.addEventListener('click', addBackdrop);
  };

  const addBackdrop = (e: React.MouseEvent) => {
    const div = document.getElementById('c5-date-picker');
    const node = ReactDOM.findDOMNode(div);
    // @ts-ignore
    if (showDatePicker && !node?.contains(e.target)) {
      setShowDatePicker(false);
    }
  };

  const handleMonthChange = (increment: number) => {
    let month = defaultDate.getMonth() + 1 + increment;
    let year = defaultDate.getFullYear();
    console.log('month', month);
    console.log('year', year);
    const day = defaultDate.getDate();
    if (month === 0) {
      month = 12;
      year--;
    } else if (month > 12) {
      year++;
      month = 1;
    }

    const newDate = convertToDateFormat(
      `${year.toString()}-${
        month.toString().length === 1
          ? '0' + month.toString()
          : month.toString()
      }-${day.toString().length === 1 ? '0' + day.toString() : day.toString()}`
    );
    setDefaultDate(new Date(newDate));
  };

  const handleYearChange = (increment: number) => {
    const month = defaultDate.getMonth() + 1;
    const year = defaultDate.getFullYear() + increment;
    const day = defaultDate.getDate();
    const newDate = convertToDateFormat(
      `${year.toString()}-${
        month.toString().length === 1
          ? '0' + month.toString()
          : month.toString()
      }-${day.toString().length === 1 ? '0' + day.toString() : day.toString()}`
    );
    setDefaultDate(new Date(newDate));
  };

  type CalendarDay = {
    date: number;
    day: number;
    dayString: string;
    month: number;
    timestamp: number;
  };

  const isCurrentDay = (day: CalendarDay) => {
    if (day.month === 0 && day.date === defaultDate.getDate()) {
      return true;
    } else {
      return false;
    }
  };

  const isSelectedDay = (day: CalendarDay) => {
    return day.timestamp === defaultDate.getTime();
  };

  const onDateClick = (day: any) => {
    const newDate = new Date(day.timestamp);
    setDefaultDate(newDate);
    if (props.onChange) {
      props.onChange(newDate);
    }
    setShowDatePicker(false);
  };

  const renderCalendar = () => {
    if (monthDetails.length === 0) {
      return <div>There are no results</div>;
    }
    let days = monthDetails.map((day: any, index: number) => {
      return (
        <div
          className={`c-day-container ${day.month !== 0 ? 'disabled' : ''} ${
            isCurrentDay(day) ? ' highlight' : ''
          } ${isSelectedDay(day) ? ' hightlight-green' : ''}`}
          key={index}
        >
          <div className="cdc-day">
            <span onClick={() => onDateClick(day)}>{day.date}</span>
          </div>
        </div>
      );
    });

    return (
      <div className="c-container">
        <div className="cc-head">
          {['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'].map((d, i) => (
            <div key={i} className="cch-name">
              {d}
            </div>
          ))}
        </div>
        <div className="cc-body">{days}</div>
      </div>
    );
  };

  return (
    <div id="c5-date-picker" className="c5-date-picker">
      <div className="c5-dp-input" onClick={handleDateClick}>
        <input type="date" ref={dateRef} />
      </div>
      {showDatePicker ? (
        <div className="mdp-container">
          <div className="mdp-content-container">
            <div className="mdp-head">
              <div className="mdp-button">
                <div
                  className="mdp-head-button-inner"
                  onClick={() => handleYearChange(-1)}
                >
                  <Tooltip
                    position="bottom"
                    message="Subtract Year"
                    theme="dark"
                  >
                    <span className="mdp-head-button-left-arrows"></span>
                  </Tooltip>
                </div>
              </div>
              <div className="mdp-button">
                <div
                  className="mdp-head-button-inner"
                  onClick={() => handleMonthChange(-1)}
                >
                  <Tooltip
                    position="bottom"
                    message="Subtract month"
                    theme="dark"
                  >
                    <span className="mdp-head-button-left-arrow"></span>
                  </Tooltip>
                </div>
              </div>
              <div className="mdp-head-date">
                <div className="mdp-head-year">{defaultDate.getFullYear()}</div>
                <div className="mdp-head-month">
                  {months[defaultDate.getMonth()]}
                </div>
              </div>
              <div className="mdp-button">
                <div
                  className="mdp-head-button-inner"
                  onClick={() => handleMonthChange(1)}
                >
                  <Tooltip position="bottom" message="Add month" theme="dark">
                    <span className="mdp-head-button-right-arrow"></span>
                  </Tooltip>
                </div>
              </div>
              <div className="mdp-button">
                <div
                  className="mdp-head-button-inner"
                  onClick={() => handleYearChange(1)}
                >
                  <Tooltip position="bottom" message="Add year" theme="dark">
                    <span className="mdp-head-button-right-arrows"></span>
                  </Tooltip>
                </div>
              </div>
            </div>
          </div>

          <div className="mdp-body">{renderCalendar()}</div>
        </div>
      ) : null}
    </div>
  );
};

export default DatePicker;
