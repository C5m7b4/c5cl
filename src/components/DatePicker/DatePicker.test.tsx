import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import DatePicker from './DatePicker';

describe('DatePicker', () => {
  test('should render correctly', () => {
    const testFn = jest.fn();
    const { container } = render(
      <DatePicker onChange={testFn} date={new Date('1/1/2022')} />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle year change', () => {
    const testFn = jest.fn();
    const { container, getByRole } = render(
      <DatePicker onChange={testFn} date={new Date('1/1/2022')} id="1" />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    const btn1 = container.querySelector('#c5-dp-1-ly') as HTMLDivElement;
    fireEvent.click(btn1);
    const btn2 = container.querySelector('#c5-dp-1-ny') as HTMLElement;
    fireEvent.click(btn2);
  });
  test('should handle month change', () => {
    const testFn = jest.fn();
    const { container, getByRole } = render(
      <DatePicker onChange={testFn} date={new Date('1/1/2022')} id="1" />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    const btn1 = container.querySelector('#c5-dp-1-lm') as HTMLDivElement;
    fireEvent.click(btn1);
    const btn2 = container.querySelector('#c5-dp-1-nm') as HTMLDivElement;
    fireEvent.click(btn2);
  });
  test('should handle the year changing', () => {
    const testFn = jest.fn();
    const { container, getByRole } = render(
      <DatePicker onChange={testFn} date={new Date('12/31/2021')} id="1" />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    const btn2 = container.querySelector('#c5-dp-1-nm') as HTMLDivElement;
    fireEvent.click(btn2);
  });
  test('should handle a new date click', () => {
    const testFn = jest.fn();
    const { container, getByText } = render(
      <DatePicker onChange={testFn} date={new Date('12/31/2021')} id="1" />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);

    const day = getByText('17');
    fireEvent.click(day);
  });
});
