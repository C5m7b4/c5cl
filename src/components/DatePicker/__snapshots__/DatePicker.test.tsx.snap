// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render correctly 1`] = `
<div>
  <div
    class="c5-date-picker"
    id="c5-date-picker"
  >
    <div
      class="c5-dp-input"
      id="c5-dp-1"
    >
      <input
        type="date"
      />
    </div>
    <div
      style="position: fixed; z-index: 1; top: 0px; bottom: 0px; left: 0px; right: 0px; display: none; background: rgba(0, 0, 0, 0.3); cursor: pointer;"
    />
    <div
      class="color-picker-modal-wrapper"
      style="position: fixed; top: 50%; left: 50%; z-index: 2; transform: translate(-50%, -50%); transform-origin: left top; max-width: 100%; height: auto; display: none; animation: color-picker-fade-in .5s ease;"
    >
      <div
        class="mdp-container"
        style="position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="mdp-content-container"
        >
          <div
            class="mdp-head"
          >
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
                id="c5-dp-1-ly"
              >
                <span
                  class="mcl-tooltip-wrapper"
                >
                  <div
                    class="mcl-tooltip-bubble mcl-tooltip-bottom dark"
                  >
                    <div
                      class="mcl-tooltip-message"
                      style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255); padding: 3px;"
                    >
                      Subtract Year
                    </div>
                  </div>
                  <span
                    class="mcl-tooltip-trigger"
                  >
                    <span
                      class="mdp-head-button-left-arrows"
                    />
                  </span>
                </span>
              </div>
            </div>
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
                id="c5-dp-1-lm"
              >
                <span
                  class="mcl-tooltip-wrapper"
                >
                  <div
                    class="mcl-tooltip-bubble mcl-tooltip-bottom dark"
                  >
                    <div
                      class="mcl-tooltip-message"
                      style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255); padding: 3px;"
                    >
                      Subtract month
                    </div>
                  </div>
                  <span
                    class="mcl-tooltip-trigger"
                  >
                    <span
                      class="mdp-head-button-left-arrow"
                    />
                  </span>
                </span>
              </div>
            </div>
            <div
              class="mdp-head-date"
            >
              <div
                class="mdp-head-year"
              >
                2022
              </div>
              <div
                class="mdp-head-month"
              >
                January
              </div>
            </div>
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
                id="c5-dp-1-nm"
              >
                <span
                  class="mcl-tooltip-wrapper"
                >
                  <div
                    class="mcl-tooltip-bubble mcl-tooltip-bottom dark"
                  >
                    <div
                      class="mcl-tooltip-message"
                      style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255); padding: 3px;"
                    >
                      Add month
                    </div>
                  </div>
                  <span
                    class="mcl-tooltip-trigger"
                  >
                    <span
                      class="mdp-head-button-right-arrow"
                    />
                  </span>
                </span>
              </div>
            </div>
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
                id="c5-dp-1-ny"
              >
                <span
                  class="mcl-tooltip-wrapper"
                >
                  <div
                    class="mcl-tooltip-bubble mcl-tooltip-bottom dark"
                  >
                    <div
                      class="mcl-tooltip-message"
                      style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255); padding: 3px;"
                    >
                      Add year
                    </div>
                  </div>
                  <span
                    class="mcl-tooltip-trigger"
                  >
                    <span
                      class="mdp-head-button-right-arrows"
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mdp-body"
        >
          <div
            class="c-container"
          >
            <div
              class="cc-head"
            >
              <div
                class="cch-name"
              >
                Sun
              </div>
              <div
                class="cch-name"
              >
                Mon
              </div>
              <div
                class="cch-name"
              >
                Tue
              </div>
              <div
                class="cch-name"
              >
                Wed
              </div>
              <div
                class="cch-name"
              >
                Thu
              </div>
              <div
                class="cch-name"
              >
                Fri
              </div>
              <div
                class="cch-name"
              >
                Sat
              </div>
            </div>
            <div
              class="cc-body"
            >
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    30
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   highlight  hightlight-green"
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    1
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    2
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    3
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    4
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    5
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    6
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    7
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    8
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    9
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    10
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    11
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    12
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    13
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    14
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    15
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    16
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    17
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    18
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    19
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    20
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    21
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    22
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    23
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    24
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    25
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    26
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    27
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    28
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    29
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    30
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    31
                  </span>
                </div>
              </div>
              <div
                class="c-day-container   highlight  hightlight-green"
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    1
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    2
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    3
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    4
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    5
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    6
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    7
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    8
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    9
                  </span>
                </div>
              </div>
              <div
                class="c-day-container disabled  "
              >
                <div
                  class="cdc-day"
                >
                  <span>
                    10
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
